test/e2e/e2e_test.go

```diff
@@ -483,6 +485,100 @@ var _ = ginkgo.Describe("Kueue visibility server", func() {
 			})
 		})
 	})
+
+	ginkgo.When("A subject is bound to kueue-batch-admin-role", func() {
+		var clusterRoleBinding *rbacv1.ClusterRoleBinding
+
+		ginkgo.BeforeEach(func() {
+			clusterRoleBinding = &rbacv1.ClusterRoleBinding{
+				ObjectMeta: metav1.ObjectMeta{Name: "read-pending-workloads"},
+				RoleRef:    rbacv1.RoleRef{APIGroup: rbacv1.GroupName, Kind: "ClusterRole", Name: "kueue-batch-admin-role"},
+				Subjects: []rbacv1.Subject{
+					{Name: "default", APIGroup: "", Namespace: "kueue-system", Kind: rbacv1.ServiceAccountKind},
+				},
+			}
+			gomega.Expect(k8sClient.Create(ctx, clusterRoleBinding)).Should(gomega.Succeed())
+		})
+
+		ginkgo.AfterEach(func() {
+			gomega.Expect(k8sClient.Delete(ctx, clusterRoleBinding)).To(gomega.Succeed())
+		})
+
+		ginkgo.It("Should give access to pending workloads in ClusterQueue", func() {
+			ginkgo.By("Verify the ClusterQueue does not exist", func() {
+				_, err := impersonatedVisibilityClient.ClusterQueues().GetPendingWorkloadsSummary(ctx, "non-existent", metav1.GetOptions{})
+				statusErr, ok := err.(*errors.StatusError)
+				gomega.Expect(ok).To(gomega.Equal(true))
+				gomega.Expect(statusErr.ErrStatus.Reason).To(gomega.Equal(metav1.StatusReasonNotFound))
+			})
+		})
+
+		ginkgo.It("Should give access to pending workloads in LocalQueue", func() {
```

Comment:
I think it does not seem to require a dedicated `it`, because it costs running the BeforeEach.